### PR TITLE
CPS / Get quote details data

### PR DESCRIPTION
**Pull request by styrelius**
## What?
Using the new generic `data` property on quotes instead of `quoteDetails`, and querying for exactly that where we need no more data than that.

- Types for the different `data` objects we expect to get on the different quotes, in `quoteDetailsDataTypes.ts`
- New `QuoteDetailsData` query, and generated hooks for that, to be able to make _one_ request to only get the `data`
- `useQuoteCartIdFromUrl` hook so that we don't have to do the `useParams<{ id: string }>()` everywhere
- `QuoteDetails` component remade to not get its data as props but fetch is directly in the component
- `getQuoteDetails.ts` refactored
- Unit tests for `getQuoteDetails` function. These are pretty suboptimal though, and it would make more sense IMO to at some point replace these with component tests with React Testing Library
- A few new text keys and translations for quote details labels and values
- Quote details Storybook story is modified to work with the new data fetching structure for `QuoteDetails` (this includes adding some mocked responses to the new `QuoteDetailsData` query)

## Why?

- The generic `data` object on `quote`s replaces the now deprecated `quoteDetails` property (reason is making the quotes generic on the back-end, an IPC project), which means we don't get any types from our schema telling us what quote details properties we can expect. Hence types being declared here.
- Using the `data` directly from the `QuoteDetails` component, with a query of its own, follows the [Apollo best practices](https://www.apollographql.com/docs/react/data/operation-best-practices/#query-only-the-data-you-need-where-you-need-it) that we've been trying to follow with the new checkout components 

**Ticket(s): [GRW-730]**

## Screenshots / recordings 

**Swedish House Quote:**
image

**Norwegian Combo Quote:**
image

**Danish Home Contents Student:**
image

## How to test
Check out this branch and run Storybook to view the `QuoteDetails` component with mocked data.

[GRW-730]: https://hedvig.atlassian.net/browse/GRW-730?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/pages/Offer/Introduction/Sidebar/index.tsx [hunk 1]
```diff
@@ -131,7 +131,7 @@ export const Sidebar: React.FC<SidebarProps> = ({
   campaign,
   onCheckoutOpen,
 }) => {
-  const { id: quoteCartId } = useParams<{ id: string }>()
+  const { quoteCartId } = useQuoteCartIdFromUrl()
```
Comment (dendoesit) [inline src/client/pages/Offer/Introduction/Sidebar/index.tsx:134]: nice scouting 👍

Comment (dendoesit) [inline src/client/pages/Offer/Introduction/Sidebar/index.tsx:134]: nice scouting 👍

File: src/client/data/graphql.tsx [hunk 2]
```diff
@@ -9870,6 +9890,17 @@ export type StartSignResponse =
...
+  __typename?: 'StoredCardDetails'
+  id: Scalars['String']
+  cardName?: Maybe<Scalars['String']>
+  brand?: Maybe<Scalars['String']>
+  lastFourDigits: Scalars['String']
+  expiryMonth: Scalars['String']
+  expiryYear: Scalars['String']
+  holderName?: Maybe<Scalars['String']>
+}
+
```
[the remark this answers is not in the document]
Comment (styrelius) [reply src/client/data/graphql.tsx:9903]: This is just generated from our GraphQL schema so I don't know, ask the MX team 🙃 (But card payments are connected to Adyen and not Trustly)

File: src/client/pages/Offer/Introduction/Sidebar/index.tsx (hunk 1 above)
Comment (styrelius) [reply src/client/pages/Offer/Introduction/Sidebar/index.tsx:134]: Thank you 🙏

**Review by dendoesit — APPROVED** [review]
great work Ellen 🎉

**Comment by styrelius** [comment]
> great work Ellen 🎉

Did it together with Amanda 😊

**Review by dendoesit — APPROVED** [review]
👍

**Pull request merged**